### PR TITLE
add program exception model

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

adds mongoose schema for program exceptions and methods to interact with db

**Type of Change**

- [ ] Bug
- [ ] Refactor
- [x] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [x] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [x] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

**Comment by joneubank** [comment]
Can we rename the file to `exceptions-repo`? just to prevent us having a million files all named `repo` (plus this follows the pattern established in the submission code).

File: src/exception/repo.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  find(name: string): Promise<DeepReadonly<ProgramException> | undefined>;
+  update(programException: ProgramException): Promise<DeepReadonly<ProgramException> | undefined>;
+  delete(name: string): Promise<void>;
+}
+
+export const programExceptionRepository: ProgramExceptionRepository = {
+  async create(req: ProgramException) {
+    L.debug(`Creating new program exception with: ${JSON.stringify(req)}`);
+    const exception: ProgramException = { ...req };
+    const programException = new ProgramExceptionModel(exception);
```
Comment (joneubank) [inline src/exception/repo.ts:66]: Why not just pass your argument directly to the model?
```suggestion
    const programException = new ProgramExceptionModel(req);
```

File: src/exception/repo.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        return F(MongooseUtils.toPojo(doc) as ProgramException);
+      }
+    } catch (e) {
+      L.error('failed to find program exception', e);
+      throw new Error(`failed to find program exception with name: ${JSON.stringify(name)}`);
+    }
+  },
+
+  // update replaces entire program exceptions record
+  async update(programException: ProgramException) {
```
Comment (joneubank) [inline src/exception/repo.ts:93]: Let's just name this replace ;)

**Review by joneubank — CHANGES_REQUESTED** [review]
Looks good overall, just some nit picks that we can use to clean up a bit.

**Comment by ciaranschutte** [comment]
thanks for review @joneubank - addressed feedback

**Review by joneubank — APPROVED** [review]
LGTM 👍

**Pull request merged**